project-koku/koku
# [COST-1852] - Update SQL to properly calculate node capacity

## Summary
This will correctly calculate node capacity for OpenShift.

## Testing

1. Edit on prem cost model to have rates

```
{
    "name": "Cost Management OpenShift Cost Model",
    "description": "A cost model of on-premises OpenShift clusters.",
    "source_type": "OCP",
    "source_uuids": ["PROVIDER_UUID"],
    "rates": [
    {
        "metric": {"name": "node_cost_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 1000.0,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Infrastructure"
    },
    {
        "metric": {"name": "cluster_cost_per_month"},
        "tiered_rates": [{
            "unit": "USD",
            "value": 10000.0,
            "usage_start": null,
            "usage_end": null
        }],
        "cost_type": "Infrastructure"
    }
    ]
}
```
    
2. Run make load test customer data
3. Check http://127.0.0.1:8000/api/cost-management/v1/reports/openshift/costs/?filter[time_scope_value]=-1&filter[cluster]=OCP%20on%20Premises&group_by[node]=* and confirm that each node has the same cost, since each has the same number of CPUs. 

[Smoke Tests](https://ci.ext.devshift.net/job/project-koku-koku-pr-check/1347/console)

```
= 5814 passed, 196 skipped, 618 deselected, 26 warnings in 3970.97s (1:06:10) ==
```
